lib/datadog/opentelemetry/configuration/settings.rb

```diff
@@ -0,0 +1,125 @@
+# frozen_string_literal: true
+
+require 'json'
+require_relative '../../core/configuration/ext'
+
+module Datadog
+  module OpenTelemetry
+    module Configuration
+      module Settings
+        def self.extended(base)
+          base = base.singleton_class unless base.is_a?(Class)
+          add_settings!(base)
+        end
+
+        def self.json_parser(env_var_name)
+          proc do |value|
+            next {} if value.nil? || value.empty?
+            parsed = JSON.parse(value)
+            unless parsed.is_a?(Hash)
+              Datadog.logger.warn("#{env_var_name} must be a JSON object (hash), got: #{parsed.class}")
+              next {}
+            end
+            parsed
+          rescue JSON::ParserError => exc
+            Datadog.logger.warn("Failed to parse #{env_var_name}: #{exc.class}: #{exc}: #{value}")
+            {}
+          end
+        end
+
+        def self.add_settings!(base)
+          base.class_eval do
+            settings :opentelemetry do
+              settings :exporter do
+                option :protocol do |o|
+                  o.type :string
+                  o.env 'OTEL_EXPORTER_OTLP_PROTOCOL'
+                  o.default 'http/protobuf'
+                end
+
+                option :timeout_millis do |o|
+                  o.type :int
+                  o.env 'OTEL_EXPORTER_OTLP_TIMEOUT'
+                  o.default 10_000
+                end
+
+                option :headers do |o|
+                  o.type :hash
+                  o.env 'OTEL_EXPORTER_OTLP_HEADERS'
+                  o.default { {} }
+                  o.env_parser(&Settings.json_parser('OTEL_EXPORTER_OTLP_HEADERS'))
```

Comment:
The format for `OTEL_EXPORTER_OTLP_HEADERS` seems incorrect [as per OTel spec](https://opentelemetry.io/docs/languages/sdk-configuration/otlp-exporter/#otel_exporter_otlp_headers).
More specifically, the format should match the [W3C Baggage](https://www.w3.org/TR/baggage/#header-content) header format (except that additional semi-colon delimited metadata is not supported): [see OTel reference link for details](https://opentelemetry.io/docs/specs/otel/protocol/exporter/#specifying-headers-via-environment-variables).

Reply:
This method seems to do exactly what you need: https://github.com/DataDog/dd-trace-rb/blob/2c84fc089f429cc1d1547cdff4a56b41bcf3e768/lib/datadog/tracing/distributed/baggage.rb#L138